README.md

```diff
@@ -5,7 +5,128 @@
 
 ## Getting started
 
-TODO
+### What does the Workleap.EventPropagation package do?
+* Provides list of existing Topics
+* Serializes domain events
+* Manages publication fields
+* Provides webhook endpoint (.net core only)
+* Provides IDomainEvent and IDomainEventHandler<> interfaces
+* Deserializes domain events
+* Mediates call to DomainEventHandlers
+* Validates configurations
+* Handles topic subscriptions
+
+### Using the Workleap.EventPropagation package to publish events
+
+When using dotnet core, you can register event propagation at startup in the service collection.
+```
+services
+    .AddEventPropagation()
+    .Configure(this.Configuration);
+```
+Configuration is required. Configuration can be loaded from the appsettings file by passing the IConfiguration instance (see above). The topic access key should be stored securely in a key vault.
+```
+"EventPropagation": {
+    "TopicName": "...",
+    "TopicAccessKey": "...",
+    "TopicEndpoint": "http://topicurl.domain.com/api/events"
+  }
+```
+
+To publish an event, use the IEventPropagationClient interface (via injection). Use the PublishDomainEventAsync to publish the event. The required Subject field is a string description to provide context for the event.
+
+```
+var coolCandorEvent = new ExampleDomainEvent
+{
+    CoolId = Guid.NewGuid().ToString(),
+    CoolDate = DateTime.UtcNow
+};
+
+await this._eventPropagationClient.PublishDomainEventAsync(subject: "TestEventPublication", coolCandorEvent);
+```
+
+#### Restrictions
+* Your service can publish events on only 1 topic.
+* The topic the service publishes to must be in the Topics class provided by the Workleap.EventPropagation package.
+
+#### Exceptions
+
+In case of error while publishing an event, an exception of type EventPropagationPublishingException is thrown. It provides the TopicName, Subject and TopicHostName fields.
+
+If configuration is bad, an exception of type InvalidConfigurationException is thrown. If the provided topic is bad, an exception of type InvalidTopicException is thrown.
+
+### Using the Workleap.EventPropagation package to subscribe to events
+
+When using dotnet core, you can register event propagation subscriptions at startup in the service collection.  To configure the subscriber, the list of subscribed topics is required.
+
+```
+services
+    .AddEventPropagationSubscriber()
+    .AddDomainEventHandlersFromAssembly(Assembly.GetExecutingAssembly())
+    .Configure(new[] { Topics.Organisation, Topics.Candor });
+```
+
+An API endpoint must be made available to receive events from the EventGrid topic. The package provides this endpoint for dotnet core projects. It can be added like so.
+```
+services.AddControllers().PartManager.ApplicationParts.Add(new AssemblyPart(typeof(EventGridController).Assembly));
```

Comment:
Think this is the 'old' way of doing it, now we have the AddEventPropagationEndpoints extension method to register those endpoints

Reply:
This comment is still relevant I think

Reply:
Yep @vad3245 in order to write the right documentation you should try the library